tests/unit/amo/components/TestUserProfile.js

```diff
@@ -0,0 +1,318 @@
+        userId: 500,
+        username: 'tofumatt',
+      },
+    }).store,
+    ...props
+  // eslint-disable-next-line padded-blocks
+  } = {}) {
+
+    return shallowUntilTarget(
+      <UserProfile i18n={i18n} params={params} store={store} {...props} />,
+      UserProfileBase,
+    );
+  }
+
+  it('renders user profile page', () => {
+    const root = renderUserProfile();
+
+    expect(root.find('.UserProfile')).toHaveLength(1);
+  });
+
+  it('dispatches fetchUserAccount action if username is not found', () => {
+    const { store } = dispatchSignInActions();
+    const dispatchSpy = sinon.spy(store, 'dispatch');
+    const username = 'i-am-not-in-the-store';
+
+    const root = renderUserProfile({ params: { username }, store });
+
+    sinon.assert.calledWith(dispatchSpy, fetchUserAccount({
+      errorHandlerId: root.instance().props.errorHandler.id,
+      username,
+    }));
+  });
+
+  it('dispatches fetchUserAccount action if username param changes', () => {
+    const { store } = dispatchSignInActions({ username: 'black-panther' });
+    const dispatchSpy = sinon.spy(store, 'dispatch');
+
+    const root = renderUserProfile({
+      params: { username: 'black-panther' },
+      store,
+    });
+
+    dispatchSpy.reset();
+
+    sinon.assert.notCalled(dispatchSpy);
+
+    root.setProps({ params: { username: 'killmonger' } });
+
+    sinon.assert.calledWith(dispatchSpy, fetchUserAccount({
+      errorHandlerId: root.instance().props.errorHandler.id,
+      username: 'killmonger',
+    }));
+  });
+
+  it('does not dispatch fetchUserAccount if username does not change', () => {
+    const { store } = dispatchSignInActions({ username: 'black-panther' });
+    const dispatchSpy = sinon.spy(store, 'dispatch');
+
+    const root = renderUserProfile({
+      params: { username: 'black-panther' },
+      store,
+    });
+
+    dispatchSpy.reset();
+
+    sinon.assert.notCalled(dispatchSpy);
+
+    root.setProps({ params: { username: 'black-panther' } });
+
+    sinon.assert.notCalled(dispatchSpy);
+  });
+
+  it('renders the user avatar', () => {
+    const root = renderUserProfile();
+
+    const headerContent = root.find('.UserProfile-user-info').prop('header');
+    const header = shallow(<div>{headerContent}</div>);
+
+    expect(header.find(UserAvatar)).toHaveProp(
+      'user', getCurrentUser(root.instance().props.store.getState().users));
+  });
+
+  it('still passes user prop to avatar while loading', () => {
+    const root = renderUserProfile({ params: { username: 'not-ready' } });
+
+    const headerContent = root.find('.UserProfile-user-info').prop('header');
+    const header = shallow(<div>{headerContent}</div>);
+
+    expect(header.find(UserAvatar)).toHaveProp('user', undefined);
+  });
+
+  it("renders the user's name", () => {
+    const root = renderUserProfile();
+
+    const headerContent = root.find('.UserProfile-user-info').prop('header');
+    const header = shallow(<div>{headerContent}</div>);
```

Comment:
These two lines are repeated a lot. You could move them to a helper like:

```js
const header = getHeaderComponent(root);
```

Reply:
Hah, I was thinking that when I refactored it and I was lazy. Thanks for calling me out on that, will fix 👍